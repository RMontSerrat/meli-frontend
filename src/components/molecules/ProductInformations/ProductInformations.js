import React from 'react';
import PropTypes from 'prop-types';
import styles from './ProductInformations.module.scss';
import { Price } from '~/components/atoms';

const ProductInformations = ({
  price, name, condition, soldQuantity,
}) => (
  <div className={styles.ProductInformations} data-testid="ProductInformations">
    <span className={styles.subtitle}>
      {condition}
      {' '}
      |
      {' '}
      {soldQuantity}
      {' '}
      vendidos
    </span>
    <h1 className={styles.title}>{name}</h1>
    <Price currency={price.currency} amount={price.amount} decimals={price.decimals} variant="big" />
  </div>
);

ProductInformations.propTypes = {
  price: PropTypes.shape({
    currency: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    decimals: PropTypes.number.isRequired,
  }).isRequired,
  name: PropTypes.string.isRequired,
  condition: PropTypes.string.isRequired,
  soldQuantity: PropTypes.number.isRequired,
};

export default ProductInformations;
