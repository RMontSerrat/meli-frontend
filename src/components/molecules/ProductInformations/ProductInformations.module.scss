.ProductInformations {
  .subtitle {
    font-family: $font-family;
    font-size: size(small);
    font-weight: $font-thin;
    white-space: pre-wrap;
  }

  .title {
    font-family: $font-family;
    font-size: size(xbig);
    font-weight: $font-bold;
    margin: 0;
    padding: rem(10px) 0 rem(20px);
  }
}
