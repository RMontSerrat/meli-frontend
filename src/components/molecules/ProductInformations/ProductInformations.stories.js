/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import ProductInformations from './ProductInformations';

storiesOf('ProductInformations', module).add('default', () => {
  const price = {
    currency: "$",
    amount: 10000,
    decimals: 11,
  }
  return (
    <ProductInformations name="Bateria P/ Note Dell Inspiron N4050 J1knd P22g" price={price} soldQuantity={122} condition="Novo" />
  )
});
