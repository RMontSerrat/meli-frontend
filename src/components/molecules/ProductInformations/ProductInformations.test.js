import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ProductInformations from './ProductInformations';

describe('<ProductInformations />', () => {
  const price = {
    currency: '$',
    amount: 10000,
    decimals: 11,
  };

  const productProps = {
    name: 'Bateria P/ Note Dell Inspiron N4050 J1knd P22g',
    price,
    soldQuantity: 122,
    condition: 'Novo',
  };

  test('it should mount', () => {
    render(<ProductInformations {...productProps} />);

    const productInformations = screen.getByTestId('ProductInformations');

    expect(productInformations).toBeInTheDocument();
  });

  test('snapshot', () => {
    const { asFragment } = render(
      <ProductInformations {...productProps} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
