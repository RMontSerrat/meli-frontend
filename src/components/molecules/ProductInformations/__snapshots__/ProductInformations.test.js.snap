// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInformations /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProductInformations"
    data-testid="ProductInformations"
  >
    <span
      class="subtitle"
    >
      Novo | 122 vendidos
    </span>
    <h1
      class="title"
    >
      Bateria P/ Note Dell Inspiron N4050 J1knd P22g
    </h1>
    <span
      class="Price big"
      data-testid="Price"
    >
      <span
        class="currency"
      >
        $
      </span>
      <span
        class="amount"
      >
        10.000
      </span>
      <span
        class="decimals"
      >
        11
      </span>
    </span>
  </div>
</DocumentFragment>
`;
